packetbeat/protos/amqp/amqp.go

```diff
@@ -346,44 +343,42 @@ func (amqp *Amqp) expireTransaction(trans *AmqpTransaction) {
 
 //This method handles published messages from clients. Being an async
 //process, the method, header and body frames are regrouped in one transaction
-func (amqp *Amqp) handlePublishing(client *AmqpMessage) {
+func (amqp *amqpPlugin) handlePublishing(client *amqpMessage) {
 
-	tuple := client.TCPTuple
+	tuple := client.tcpTuple
 	trans := amqp.getTransaction(tuple.Hashable())
 
 	if trans == nil {
-		trans = &AmqpTransaction{Type: "amqp", tuple: tuple}
-		amqp.transactions.Put(client.TCPTuple.Hashable(), trans)
+		trans = &amqpTransaction{tuple: tuple}
+		amqp.transactions.Put(client.tcpTuple.Hashable(), trans)
 	}
 
-	trans.ts = client.Ts
-	trans.Ts = client.Ts.UnixNano() / 1000
```

Comment:
The fields removed have never been read from. They are not part of the event being published or anything => save some bytes and drop unused structure fields.